src/api/libcellml/componententity.h

```diff
@@ -72,14 +102,30 @@ class LIBCELLML_EXPORT ComponentEntity: public ImportedEntity
     /**
      * @brief Remove the component with the given name.
      * Remove the first component found that matches the name @p name.  If the
-     * name is not found no component is removed.
+     * name is not found throw @c std::out_of_range.
      *
      * @overload
      * @param name The name of the component to remove.
      */
     void removeComponent(const std::string &name);
 
     /**
+     * @brief Remove the component with the given pointer.
+     * Remove the first component found with the pointer @p component.  If the
+     * component is not found throw @c std::out_of_range.
+     *
```

Comment:
As above re: "overload". I can bring this up in the next round of general cleanup :)